File: src/main/java/ru/market/pricing_service/model/entity/Discount.java
package ru.market.pricing_service.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Discount {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer discountId;

    @Column(length = 100)
    private String discountType;

    @Column(precision = 10, scale = 2)
    private BigDecimal discountSize;
}

File: src/main/java/ru/market/pricing_service/model/entity/Discount.kt
package ru.market.pricing_service.model.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import java.math.BigDecimal

@Entity
data class Discount(
    @Id
    @GeneratedValue
    val discountId: Int,

    @Column(length = 100) val discountType: String,
    @Column(precision = 10, scale = 2) val discountSize: BigDecimal
)